### PR TITLE
[SPARK-42615][CONNECT][FOLLOW-UP] Implement correct version API in SparkSession for Scala client

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Following up on https://github.com/apache/spark/pull/40210, add correct `version` in the scala client side.

### Why are the changes needed?

API coverage

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### CODE REVIEW

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -63,7 +62,9 @@ class SparkSession(
 
   private[this] val allocator = new RootAllocator()
 
-  def version: String = SPARK_VERSION
+  def version: String = {
```
Comment (hvanhovell) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala:65]: Make it a lazy val?

Comment (amaliujia) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala]: done

**Review by hvanhovell — APPROVED** [review]
LGTM

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala [hunk 2]
```diff
@@ -518,6 +518,10 @@ class ClientE2ETestSuite extends RemoteSparkSession {
     assert(!spark.conf.isModifiable("spark.sql.globalTempDatabase"))
     intercept[Exception](spark.conf.set("spark.sql.globalTempDatabase", "/dev/null"))
   }
+
+  test("SparkVersion") {
+    assert(!spark.version.isEmpty)
```
Comment (dongjoon-hyun) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala:523]: Does this fail on the `master` branch previously?

Comment (amaliujia) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala:523]: This in the past load the version directly from core, which was ok given that Spark Connect is depending on the core already. This change is to make that the API does not directly load from core but through the server, which is a long term solution.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM,

**Comment by dongjoon-hyun** [comment]
Merged to master/3.4. Thank you, @amaliujia and @hvanhovell .

**Comment by zhengruifeng** [comment]
Late LGTM, in the previous PR I am not very sure about the `session.version` so didn't touch it

**Pull request closed without merging**